### ISSUE TITLE
Connections ID's are not required to appear random

### ISSUE THREAD

**Issue by mikkelfj** [quicwg/base-drafts#1740]
When sending a new connection ID it is understood that this is to prevent linkability, but it is not required that the actual ID is hard to correlate to previous ID's.

It may be tempting to use a random router prefix and a local counter in the ID.

If a client migrates the client can choose a strong new ID, but if the server responds with a weak ID in the opposite direction, that isn't very helpful. Since the client cannot influence that, servers could be coerced into using weak ID's for monitoring purposes without actually violating the protocol.

At the same time it is necessary to encode useful routing information and therefore the CID must be easy to decode.

In conclusion, at least some guidance is needed on how to construct connection ID's.

**Comment by huitema** [comment]
@mikkelfj you cannot win that particular battle. If the server collaborates with the tracker it can provide connection IDs that appear random but are in fact the encryption of the weak ID with a key known to the monitoring service. The best you can do is develop the argument and put it in the security considerations.

**Comment by martinthomson** [comment]
Happy to take a PR (if we don't have text on this, we should).

**Comment by mikkelfj** [comment]
Note that the PR mentions correlation across connections as well as paths.
I would have said that random CID is only required for TP `disable_migration`, but that does not address different connections.

**Comment by mirjak** [comment]
We also have an issue on this for the ops-drafts (29).
